ext/semian/sysv_shared_memory.c

```diff
@@ -0,0 +1,54 @@
+#include "sysv_shared_memory.h"
+
+#include "util.h"
+
+#define TIMEOUT_MS (5 * 1e6)
+#define WAIT_MS (10)
+#define RETRIES (TIMEOUT_MS / WAIT_MS)
+
+void*
+wait_for_shared_memory(uint64_t key)
+{
+  for (int i = 0; i < RETRIES; ++i) {
+    int shmid = shmget(key, SHM_DEFAULT_SIZE, SHM_DEFAULT_PERMISSIONS);
```

Comment:
The first try uses `IPC_CREAT | IPC_EXCL`. If that fails, then we assume it's because another process has created the segment. I think my intention was to check if that other process had run the `shared_memory_init_fn` but I've tried to optimize that out by building data structures that reset to zero'ed out memory.

I removed the wait loop - we can add that feature if we need it.